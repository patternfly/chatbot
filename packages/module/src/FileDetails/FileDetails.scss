.pf-chatbot__code-language {
  color: var(--pf-t--global--text--color--subtle);
  display: flex; // this is needed for when FileDetailsLabel is clickable
  font-size: var(--pf-t--global--icon--size--font--xs);
  font-weight: var(--pf-t--global--font--weight--body--bold);
}

.pf-chatbot__code-icon {
  color: var(--pf-t--global--icon--color--status--custom--default);
  width: 24px;
  height: 24px;
}

.pf-chatbot__image-icon {
  color: var(--pf-t--global--icon--color--status--info--default);
  width: 24px;
  height: 24px;
}

.pf-chatbot__code-fileName {
  font-size: var(--pf-t--global--font--size--body--default);
}

.pf-chatbot__code-file-size {
  color: var(--pf-t--global--text--color--subtle);
}

// This is used in demos only
.pf-chatbot__file-details-example {
  background: var(--pf-t--global--background--color--secondary--default);
  height: 100px;
  display: flex;
  justify-content: center;
  align-items: center;
}
