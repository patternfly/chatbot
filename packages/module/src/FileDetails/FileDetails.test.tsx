import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import FileDetails from './FileDetails';

describe('FileDetails', () => {
  it('should render file details', () => {
    const { container } = render(<FileDetails fileName="test.txt" />);
    expect(container).toMatchSnapshot();
  });

  it('should render file details correctly if an extension we support is passed in', () => {
    render(<FileDetails fileName="test.txt" languageTestId="language" />);
    expect(screen.getByText('test')).toBeTruthy();
    expect(screen.queryByText('test.txt')).toBeFalsy();
    expect(screen.getByText('TEXT')).toBeTruthy();
    expect(screen.getByTestId('language')).toBeTruthy();
  });
  it('should skip language if we do not support an extension', () => {
    render(<FileDetails fileName="test.joke" languageTestId="language" />);
    expect(screen.getByText('test')).toBeTruthy();
    expect(screen.queryByTestId('language')).toBeFalsy();
  });
  it('should support image formats by rendering extension differently', () => {
    render(<FileDetails fileName="test.svg" languageTestId="language" />);
    expect(screen.getByText('test')).toBeTruthy();
    expect(screen.queryByText('test.svg')).toBeFalsy();
    expect(screen.queryByTestId('language')).toBeFalsy();
  });
  it('should handle truncation differently', () => {
    render(<FileDetails fileName="test.svg" languageTestId="language" hasTruncation={false} />);
    expect(screen.getByText('test.svg')).toBeTruthy();
    expect(screen.queryByTestId('language')).toBeFalsy();
  });
  it('should include file size if prop passed in', () => {
    render(<FileDetails fileName="test.joke" languageTestId="language" fileSize="100MB" />);
    expect(screen.getByText('100MB')).toBeTruthy();
  });
});
