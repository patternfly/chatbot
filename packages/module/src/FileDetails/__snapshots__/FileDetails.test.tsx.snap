// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileDetails should render file details 1`] = `
<div>
  <div
    class="pf-v6-l-flex pf-m-gap-sm pf-chatbot__file-details "
  >
    <div
      class="pf-v6-l-flex pf-m-align-items-center pf-m-align-self-center pf-m-justify-content-center pf-chatbot__code-icon"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 4C0 1.79086 1.79086 0 4 0H20C22.2091 0 24 1.79086 24 4V20C24 22.2091 22.2091 24 20 24H4C1.79086 24 0 22.2091 0 20V4Z"
          fill="currentColor"
        />
        <g
          clip-path="url(#clip0_3280_27505)"
        >
          <path
            d="M13.8204 5.63002C13.3954 5.50752 12.9529 5.75502 12.8304 6.18002L9.63035 17.38C9.50785 17.805 9.75535 18.2475 10.1804 18.37C10.6054 18.4925 11.0479 18.245 11.1704 17.82L14.3704 6.62002C14.4929 6.19502 14.2454 5.75252 13.8204 5.63002ZM15.8354 8.63252C15.5229 8.94502 15.5229 9.45252 15.8354 9.76502L18.0679 12L15.8329 14.235C15.5204 14.5475 15.5204 15.055 15.8329 15.3675C16.1454 15.68 16.6529 15.68 16.9654 15.3675L19.7654 12.5675C20.0779 12.255 20.0779 11.7475 19.7654 11.435L16.9654 8.63502C16.6529 8.32252 16.1454 8.32252 15.8329 8.63502L15.8354 8.63252ZM8.16785 8.63252C7.85535 8.32002 7.34785 8.32002 7.03535 8.63252L4.23535 11.4325C3.92285 11.745 3.92285 12.2525 4.23535 12.565L7.03535 15.365C7.34785 15.6775 7.85535 15.6775 8.16785 15.365C8.48035 15.0525 8.48035 14.545 8.16785 14.2325L5.93285 12L8.16785 9.76502C8.48035 9.45252 8.48035 8.94502 8.16785 8.63252Z"
            fill="white"
          />
        </g>
        <defs>
          <clippath>
            <rect
              fill="white"
              height="12.8"
              transform="translate(4 5.60001)"
              width="16"
            />
          </clippath>
        </defs>
      </svg>
    </div>
    <div
      class="pf-v6-l-flex pf-m-gap-xs"
    >
      <div
        class=""
      >
        <div
          class="pf-v6-l-flex pf-m-column pf-m-gap-none"
        >
          <div
            class=""
          >
            <span
              class="pf-chatbot__code-fileName"
            >
              <span
                class="pf-v6-c-truncate"
                tabindex="0"
              >
                <span
                  class="pf-v6-c-truncate__start"
                >
                  test
                </span>
              </span>
            </span>
          </div>
          <div
            class="pf-chatbot__code-language"
          >
            TEXT
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
